<h1 style="text-align: center;">Projet_N°7_THOMPSON&TORT_MARTINEZ</h1>
<h2 style="text-align: center;">Étude d'une centrale à cycle combiné à gaz</h2>

<div style="text-align: center;">
  <img src="centrale.jpg" width="900"/>
  <p><em>Centrale thermique de Bouchain, EDF</em></p>
</div>



<h3 style="text-align: center;">Étude de la turbine à gaz</h3>
<div style="text-align: center;">
  <img src="cycle.png" width="600"/>
</div>



### 2.1/
On commence par exprimer la puissance absorbée par le compresseur et la puissance délivrée par la turbine.
$$
\dot{W}_{\text{tot}} = \dot{W}_{\text{turbine}} - \dot{W}_{\text{compresseur}}
$$

D'après le premier principe industriel entre 1-2 et entre 2-3:
$$
\dot{W}_c = \dot{m}_1 \, (h_2 - h_1)
$$

$$
\dot{W}_t = \dot{m}_3 \, (h_4 - h_3)
$$

donc:
$$
\dot{W}_c = \dot{m}_1 \, C_{p,a} \, (T_2 - T_1)
$$
$$
\dot{W}_t = \dot{m}_3 \, C_{p,f} \, (T_4 - T_3)
$$



On se place dans le cas réversible. On peut donc utiliser les lois de Laplace (Gaz parfait + adiabatique réversible)
$$
\frac{T_2}{T_1} = \left( \frac{P_2}{P_1} \right)^{\frac{\gamma - 1}{\gamma}}
$$

On a $ P_3 = P_2\ et\ P_4 = P_{atm} = P_1 $ donc on remplace $ \frac{P_2}{P_1}\ et\ \frac{P_4}{P_3}\ par\ r_c$ 
$$
\dot{W}_{c,is} = \dot{m}_1 \, C_{p,a} \, T_1 \cdot \left( r_c^{k_a} - 1 \right)
$$

$$
\dot{W}_{t,is} = \dot{m}_3 \, C_{p,f} \, T_3 \cdot \left(1 - r_c^{- \frac{1}{k_f}} \right)
$$

Dans le cas réel, en prenant en compte le rendement isentropique:

$$
\dot{W}_{c,is} = \frac{\dot{m}_1 \, C_{p,a} \, T_1 }{\eta_c} \cdot \left( r_c^{k_a} - 1 \right)
$$

$$
\dot{W}_{t,is} = \dot{m}_3 \, C_{p,f} \, T_3 \eta_t \cdot \left(1 - r_c^{- \frac{1}{k_f}} \right)
$$
La puisance totale vaut donc:

$$
\dot{W}_{\text{tot}} = \dot{m}_3 \, C_{p,f} \, T_3 \eta_t \cdot \left(1 - r_c^{- \frac{1}{k_f}} \right) - \frac{\dot{m}_1 \, C_{p,a} \, T_1 }{\eta_c} \cdot \left( r_c^{k_a} - 1 \right)
$$

Pour trouver la condition d'optimalité sur la puissance, on dérive l'expression par rapport à $ r_c $ et on l'annule/

$$
\frac{d\dot{W}_{\text{tot}}}{d r_c} =\dot{m}_3 \, C_{p,f} \, T_3 \eta_t \cdot \left( k_f \, r_c^{-k_f - 1} \right) - \frac{\dot{m}_1 \, C_{p,a} \, T_1 }{\eta_c} \cdot\left( k_a \, r_c^{k_a - 1} \right) = 0

$$


Après résolution on obtient:
$$
r_c^{k_a + k_f} =
\frac{ \dot{m}_3 \, C_{p,f} \, T_3 \cdot \eta_T \cdot k_f }{ \dot{m}_1 \, C_{p,a} \, T_1 \cdot \frac{k_a}{\eta_c} }
$$

enfin:
$$
r_c = 
\left[
\eta_c \, \eta_T \, \frac{T_3}{T_1} \,\frac{ ( \dot{m} \, C_p )_3
}{
( \dot{m} \, C_p )_1 \cdot
}
\frac{k_f}{k_a}
\right]^{\frac{1}{k_a + k_f}}
$$

On réalise le bilan dans la chambre de combustion (2->3):
$$
(\dot{m}_{3} - \dot{m}_{1}) \cdot \text{PCI} = \dot{m}_3 \cdot h_3 - \dot{m}_1 \cdot h_2 - (\dot{m}_{3} - \dot{m}_{1}) \cdot h_{17}
$$
car $\dot{m}_1 = \dot{m}_2$ et $ \dot{m}_{17} = \dot{m}_{3} - \dot{m}_{1}$

soit 
$$
\dot{m}_3 = \frac{\dot{m}_1 (C_{p,a}T_2 - C_{p,g}T_{17} - \text{PCI}) }{C_{p,f}T_3 - C_{p,g}T_{17} - \text{PCI})}
$$

$T_3$ et $T_{17}$ sont fixées donc $h_3$ dépend uniquement de $T_2$

or 
$$
T_2 = T_1 \left[ 1 + \frac{1}{\eta_c} \left( r_c^{k_a} - 1 \right) \right]
$$

Ainsi $\dot{m}_3$ dépend de $r_c$




### 2.2/
On utilise le code python suivant pour calculer de façon itérative les valeurs de $r_c$

In [26]:
rc = 18
gamma_a = 1.38
gamma_f = 1.31
cpa = 1039
cpg = 2256
cpf = 1213
t1 = 15+273
t3 = 1420+273
t17 = 15 + 273
eta = 0.9
m1 = 800
ka = (gamma_a-1)/gamma_a
kf = (gamma_f-1)/gamma_f
pci = 49500000

for i in range(100):
    t2 = t1*(1 + (rc**ka-1)/eta)
    m3 = m1*(cpa*t2-cpg*t17-pci)/(cpf*t3 - cpg*t17 - pci)
    rc = (eta**2)*((t3*m3*cpf*kf)/(t1*m1*cpa*ka))**(1/(ka+kf))
    
print(rc,m3)

27.270031749876463 820.968515082622


On obtient après 100 itérations du code ci dessus:

|      | Initialisation      | i = 100         |
|----------------|--------------|---------------|
| $r_c$  |$ 18 $     | $27.3 $          |
| $\dot{m}_3$  | $822,5 \ kg/s$    | $821,0 \ kg/s $      |

### 2.3/
On remplit le tableau d'état et le tableau de transformation pour le TAC:

$h_1 = C_{p,a}T_1 = 300 kj/kg $  

$T_2 = T_1 \left[ 1 + \frac{1}{\eta_c} \left( r_c^{k_a} - 1 \right) \right] = 449 K \ \ \ $ et $ \ \ \ h_2 = C_{p,a}T_2 = 750 kj/kg $  

$P_2 = P_3 = r_c P_1 = 22.8 Pa $ et $h_3 = C_{p,f}T_3= 300 kj/kg $ 

$T_4 =T_3 \left[ 1 - \eta_T ( 1 - r_c^{-k_f}) \right] = 628 K$ et $h_4 = C_{p,f}T_4 = 3762 kj/kg $  


$\dot{m}_{17} = \dot{m}_3 - \dot{m}_1 = 21 kg/s$ et $h_{17} = C_{p,g}T_{17} = 649 kj/kg $


| Point | Localisation          | État physique | P (bar) | T (°C) | \( x_v \) (-) | h (kJ/kg) |  $\dot{m}$  (kg/s) |
|-------|------------------------|----------------|---------|--------|---------------|------------|------------------------|
| 1     | Admission air          | Air            | 1.0   | 15     | 1            |  300       | 800                      |
| 2     | Sortie compresseur     | Air              | 27.3       | 449      | 1             |   750       | 800                     |
| 3     | Admission turbine      | Air            | 27.3      | 1420       | 1             | 7077     |    821                   |
| 4     | Sortie turbine         | Air              | 1.013      | 628      | 1             | 3762         | 821                     |
| 17     | Injection gaz        | Air              | 1.0       | 15    | 1             | 649         | 21                    |

Les transferts thermiques sont nuls car les parois sont calorifugées  

(1 → 2)/       $ \dot{W} = \dot{m}_1(h_2 - h_1)$  

(3 → 4)/       $ \dot{W} = \dot{m}_3(h_4 - h_3)$

| Composant                              | $ \sum \dot{m} h  (MW) $ | $ \dot{W}  (MW) $| $ \dot{Q} (MW) $ |
|----------------------------------------|---------------------------|--------------------|--------------------|
| Compresseur (1 → 2)                    | 360                         | 360       | 0                  |
| Chambre de combustion (2 et 17 → 3)    | 5197                         | 0                  | 5197 |
| Turbine (3 → 4)                        | -2731                        |  -2731        | 0                  |


### 2.3/  
La puissance totale vaut:  
$$
\dot{W}_{\text{tot}} = \dot{m}_3 \, C_{p,f} \, T_3 \eta_t \cdot \left(1 - r_c^{- \frac{1}{k_f}} \right) - \frac{\dot{m}_1 \, C_{p,a} \, T_1 }{\eta_c} \cdot \left( r_c^{k_a} - 1 \right)
$$
soit 
$$
\dot{W}_{\text{tot}} = 2371 MW
$$

On définit:  
$$\eta_{TAC} = \frac{-\dot{W}_T}{\dot{W}_c + \dot{m}_{17} \text{PCI}}$$

et on obtient:  
$$ \eta_{TAC}= 0.49$$

<h3 style="text-align: center;">5. Dimensionnement de la chaudière de récupération (HRSG)</h3>

### 5.1/  
Le coefficient d’échange global de chaque échangeur est défini suivant :
$$
\frac{1}{K} = \frac{1}{h_o} + \frac{D_o \cdot \ln \left( \frac{D_o}{D_i} \right)}{2 \lambda_{\text{acier}}} + \frac{D_o}{h_i \cdot D_i}
$$


On commence par calculer le coefficient interne $h_i$:  

$$ \frac{h_i \cdot D_i}{\lambda_t} = \frac{0.125 \cdot f(Re - 1000) \cdot {Pr}_t}{1+4.49\sqrt{f} \cdot ({Pr}_t^{2/3} -1)}

Avec:
$$
Re = \frac{4 \dot{m}_t}{\pi N_{\text{tubes}} D_i \mu_t}
$$
$$
f = \left[-1.8 \cdot \ln\left(\frac{6.9}{Re}\right)\right]^{-2}
$$
